### PR TITLE
cang-tidy static analysis fixes

### PR DESCRIPTION
**Pull request by Ingrater**
* Enable various clang-tidy static analysis checks
* Fix issues found by clang-tidy

Full list of issues clang-tidy found: [clang-tidy.log](https://github.com/ezEngine/ezEngine/files/11585123/clang-tidy.log)

### CODE REVIEW

**Review by jankrassnigg — APPROVED** [review]

**Review by C-Core — APPROVED** [review]

File: Code/Engine/RendererCore/AnimationSystem/AnimGraph/AnimNodes/PlaySequenceAnimNode.cpp [hunk 1]
```diff
@@ -331,8 +331,6 @@ void ezPlaySequenceAnimNode::Step(ezAnimGraph& graph, ezTime tDiff, const ezSkel
     return;
   }
 
-  ezAnimGraphPinDataLocalTransforms* pOutputTransform = graph.AddPinDataLocalTransforms();
```
Comment (jankrassnigg) [inline Code/Engine/RendererCore/AnimationSystem/AnimGraph/AnimNodes/PlaySequenceAnimNode.cpp:334]: Looks like this can be removed indeed.

**Review by jankrassnigg — APPROVED** [review]

File: Code/Engine/RendererCore/Components/Implementation/RopeRenderComponent.cpp [hunk 2]
```diff
@@ -65,7 +65,7 @@ void ezRopeRenderComponent::SerializeComponent(ezWorldWriter& inout_stream) cons
 void ezRopeRenderComponent::DeserializeComponent(ezWorldReader& inout_stream)
 {
   SUPER::DeserializeComponent(inout_stream);
-  const ezUInt32 uiVersion = inout_stream.GetComponentTypeVersion(GetStaticRTTI());
+  const ezUInt32 uiVersion = inout_stream.GetComponentTypeVersion(GetStaticRTTI()); // NOLINT: ignore dead store warning
```
Comment (jankrassnigg) [inline Code/Engine/RendererCore/Components/Implementation/RopeRenderComponent.cpp]: we do have a macro EZ_IGNORE_UNUSED

File: Code/Engine/Texture/Image/Implementation/ImageUtils.cpp [hunk 3]
```diff
@@ -657,6 +658,7 @@ inline static void FilterLine(
 static void DownScaleFastLine(ezUInt32 uiPixelStride, const ezUInt8* pSrc, ezUInt8* pDest, ezUInt32 uiLengthIn, ezUInt32 uiStrideIn, ezUInt32 uiLengthOut, ezUInt32 uiStrideOut)
 {
   const ezUInt32 downScaleFactor = uiLengthIn / uiLengthOut;
+  EZ_ASSERT_DEBUG(downScaleFactor >= 1, "Can not upscale");
```
Comment (jankrassnigg) [inline Code/Engine/Texture/Image/Implementation/ImageUtils.cpp]: Cannot or Can't

File: Code/Engine/Texture/TexConv/Implementation/AutoUsage.cpp [hunk 4]
```diff
@@ -141,9 +142,9 @@ static ezTexConvUsage::Enum DetectUsageFromImage(const ezImage& image)
     }
 
     // the average color in the image
-    sr /= uiNumPixels;
-    sg /= uiNumPixels;
-    sb /= uiNumPixels;
+    sr /= uiNumPixels; // NOLINT: not a division by zero
```
Comment (jankrassnigg) [inline Code/Engine/Texture/TexConv/Implementation/AutoUsage.cpp:145]: is the NOLINT still necessary, now that you have the assert ?

File: Code/EnginePlugins/JoltPlugin/Character/Implementation/JoltCharacterControllerComponent.cpp [hunk 5]
```diff
@@ -69,7 +69,7 @@ void ezJoltCharacterControllerComponent::SerializeComponent(ezWorldWriter& inout
 void ezJoltCharacterControllerComponent::DeserializeComponent(ezWorldReader& inout_stream)
 {
   SUPER::DeserializeComponent(inout_stream);
-  const ezUInt32 uiVersion = inout_stream.GetComponentTypeVersion(GetStaticRTTI());
+  const ezUInt32 uiVersion = inout_stream.GetComponentTypeVersion(GetStaticRTTI()); // NOLINT: ignore unused variable
```
Comment (jankrassnigg) [inline Code/EnginePlugins/JoltPlugin/Character/Implementation/JoltCharacterControllerComponent.cpp]: You could also just comment out these lines as long as the version isn't needed yet.

File: Code/Engine/Texture/TexConv/Implementation/AutoUsage.cpp (hunk 4 above)
Comment (Ingrater) [reply Code/Engine/Texture/TexConv/Implementation/AutoUsage.cpp:145]: Yes, unfortunately this seems to be a bit of a false positive.

File: Code/EnginePlugins/JoltPlugin/Actors/Implementation/JoltActorComponent.cpp [hunk 6]
```diff
@@ -56,7 +56,7 @@ void ezJoltActorComponent::SerializeComponent(ezWorldWriter& inout_stream) const
 void ezJoltActorComponent::DeserializeComponent(ezWorldReader& inout_stream)
 {
   SUPER::DeserializeComponent(inout_stream);
-  const ezUInt32 uiVersion = inout_stream.GetComponentTypeVersion(GetStaticRTTI());
+  const ezUInt32 uiVersion = inout_stream.GetComponentTypeVersion(GetStaticRTTI()); // NOLINT: ignore unused variable
```
Comment (jankrassnigg) [inline Code/EnginePlugins/JoltPlugin/Actors/Implementation/JoltActorComponent.cpp]: Could you review the existing NOLINT lines and remove the annotation where possible ?

File: Code/EnginePlugins/JoltPlugin/Constraints/Implementation/JoltConstraintComponent.cpp [hunk 7]
```diff
@@ -348,6 +348,13 @@ ezResult ezJoltConstraintComponent::FindParentBody(ezUInt32& out_uiJoltBodyID, e
       pObject = pObject->GetParent();
     }
 
+    if (pObject == nullptr)
```
Comment (jankrassnigg) [inline Code/EnginePlugins/JoltPlugin/Constraints/Implementation/JoltConstraintComponent.cpp:351]: This is not correct, a joint can attach something to the world, and thus a valid parent object is not necessary.

I don't see why this additional check should be necessary, the code below does handle all cases (though maybe Clang can't see that).

**Review by jankrassnigg — CHANGES_REQUESTED** [review]

**Review by jankrassnigg — APPROVED** [review]

**Comment by jankrassnigg** [comment]
I'm going to complete this PR (and later revert the change to the Jolt component), otherwise this will give me conflicts with my ongoing refactorings.

**Pull request merged**